workflows/src/test/kotlin/com/r3/corda/lib/ci/RequestKeyFlowTests.kt

```diff
@@ -0,0 +1,137 @@
+        charlieNode = mockNet.createPartyNode(CHARLIE_NAME)
+        alice = aliceNode.info.singleIdentity()
+        bob = bobNode.info.singleIdentity()
+        charlie = charlieNode.info.singleIdentity()
+        notary = mockNet.defaultNotaryIdentity
+
+        mockNet.startNodes()
+
+    }
+
+    @After
+    fun after() {
+        mockNet.stopNodes()
+    }
+
+    @Test
+    fun `request a new key`() {
+        // Alice requests that bob generates a new key for an account
+        val newKey = aliceNode.startFlow(RequestKeyInitiator(bob)).let {
+            it.getOrThrow()
+        }.publicKey
+
+        // Bob has the newly generated key as well as the owning key
+        val bobKeys = bobNode.services.keyManagementService.keys
+        val aliceKeys = aliceNode.services.keyManagementService.keys
+        assertThat(bobKeys).hasSize(2)
+        assertThat(aliceKeys).hasSize(1)
+
+        assertThat(bobNode.services.keyManagementService.keys).contains(newKey)
+
+        val resolvedBobParty = aliceNode.services.identityService.wellKnownPartyFromAnonymous(AnonymousParty(newKey))
+        assertThat(resolvedBobParty).isEqualTo(bob)
+    }
+
+    @Test
+    fun `request new key with a uuid provided`() {
+        // Alice requests that bob generates a new key for an account
+        val newKey = aliceNode.startFlow(RequestKeyInitiator(bob, UUID.randomUUID())).let {
+            it.getOrThrow()
+        }.publicKey
+
+        // Bob has the newly generated key as well as the owning key
+        val bobKeys = bobNode.services.keyManagementService.keys
+        val aliceKeys = aliceNode.services.keyManagementService.keys
+        assertThat(bobKeys).hasSize(2)
+        assertThat(aliceKeys).hasSize(1)
+
+        assertThat(bobNode.services.keyManagementService.keys).contains(newKey)
+
+        val resolvedBobParty = aliceNode.services.identityService.wellKnownPartyFromAnonymous(AnonymousParty(newKey))
+        assertThat(resolvedBobParty).isEqualTo(bob)
+    }
+
+    @Test
+    fun `verify a known key with another party`() {
+        // Charlie issues then pays some cash to a new confidential identity
+        val anonymousParty = AnonymousParty(charlieNode.startFlow(RequestKeyInitiator(alice)).let{
+            it.getOrThrow()
+        }.publicKey)
+
+        val issueTx = charlieNode.startFlow(
+                IssueTokens(listOf(1000 of USD issuedBy charlie heldBy AnonymousParty(anonymousParty.owningKey)))
+        ).getOrThrow()
+        val confidentialIdentity = issueTx.tx.outputs.map { it.data }.filterIsInstance<FungibleToken>().single().holder
+
+        // Verify Bob cannot resolve the CI before we create a signed mapping of the CI key
+        assertNull(bobNode.transaction { bobNode.services.identityService.wellKnownPartyFromAnonymous(confidentialIdentity) })
+
+        // Request a new key mapping for the CI
+        bobNode.startFlow(RequestKeyInitiator(alice, confidentialIdentity.owningKey)).let {
```

Comment:
Im wondering here whether we should split out the part of `RequestKeyInitiator` that takes an existing Key, into a new flow called something like `VerifyAndAddKey` - what do you think?